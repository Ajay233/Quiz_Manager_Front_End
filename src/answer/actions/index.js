import { getUsingParams, post, put, del } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { hideModal } from '../../modal/actions'
import { sessionExpired } from '../../utils/session'
import history from '../../history'

// create
export const addAnswer = (data, jwt) => {
  return (dispatch) => {
    return post("answer/create", [data], jwt).then((response) => {
      dispatch({
        type: "ADD_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer created", "success", true));
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        console.log(error.response);
        dispatch(setNotification("Error - unable to create answer", "error", true));
      }
    });
  }
}

// read
export const getAnswers = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("answer/findByQuestionId", param, jwt).then((response) => {
      dispatch({
        type: "SET_ANSWERS",
        payload: response.data
      });
    }).catch((error) => {
      dispatch(setNotification("Error retrieving answers", "error", true));
    });
  }
}

// update (there was a call to setCurrentAnswer but have removed this as it's not needed)
export const updateAnswer = (body, jwt) => {
  return (dispatch) => {
    return put("answer/update", [body], jwt).then((response) => {
      dispatch({
        type: "UPDATE_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer updated", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(setNotification("Error - unable to update answer", "error", true));
      }
    });
  }
}

// delete
export const deleteAnswer = (config, jwt) => {
  return (dispatch) => {
    return del("answer/delete", config, jwt).then((response) => {
      dispatch(hideModal())
      dispatch({
        type: "DELETE_ANSWER",
        payload: config.data[0]
      })
      dispatch(setNotification("Answer deleted", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(hideModal())
        dispatch(setNotification("Error - Unable to delete this answer", "error", true))
      }
    });
  }
}

export const setCurrentAnswer = (answer) => {
  return {
    type: "SET_CURRENT_ANSWER",
    payload: answer
  }
}
