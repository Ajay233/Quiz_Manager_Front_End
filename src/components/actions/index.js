export const setToTopButton = (show=false) => {
  return {
    type: "SHOW_TO_TOP_BUTTON",
    payload: show
  }
}

export const setUrl = (url="") => {
  return {
    type: "SET_URL",
    payload: url
  }
}

export const enableLightMode = () => {
  return {
    type: "SET_THEME",
    payload: false
  }
}

export const enableDarkMode = () => {
  return {
    type: "SET_THEME",
    payload: true
  }
}

export const setLoaderState = (show=false, message="", label="") => {
  return {
    type: "SET_LOADER_STATE",
    payload: {
      show: show,
      message: message,
      label: label
    }
  }
}
