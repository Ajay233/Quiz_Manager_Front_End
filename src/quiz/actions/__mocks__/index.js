export const setQuiz = jest.fn()
export const createQuiz = jest.fn( () => { return {type: "SET_QUIZ", payload:""} })
export const updateQuiz = jest.fn( () => { return {type: "SET_QUIZ", payload:""} })
export const updateQuizStatus = jest.fn()
