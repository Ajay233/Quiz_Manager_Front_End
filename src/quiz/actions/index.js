import { get, getUsingParams } from '../../axiosRequests/requests';

export const setQuizes = (quizes) => {
  return {
    type: "SET_QUIZES",
    payload: quizes
  }
}

export const getAllQuizes = (endpoint, data, token) => {
  return (dispatch) => {
    get(endpoint, data, token).then((response) => {
      console.log(response)
      dispatch({
        type: "GET_ALL_QUIZES",
        payload: response.data
      })
    }).catch((error) => {
      console.log(error.response)
    });
  }
}

export const getQuizSearchResults = (endpoint, data, token, setNotification, errorMsg) => {
  return (dispatch) => {
    getUsingParams(endpoint, data, token).then((response) => {
      dispatch({
        type: "SET_QUIZ_SEARCH_RESULTS",
        payload: response.data
      })
    }).catch((error) => {
      console.log(error.response);
      setNotification(errorMsg, "error", true);
    })
  }
}
