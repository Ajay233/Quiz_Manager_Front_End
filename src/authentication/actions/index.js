import { post } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import history from '../../history'

export const setUser = (endpoint, loginDetails) => {

  return (dispatch) => {

    post(endpoint, loginDetails).then((response) => {
      dispatch({
        type: "SET_USER_LOGGED_IN",
        payload: {
          id: response.data.user.id,
          forename: response.data.user.forename,
          surname: response.data.user.surname,
          email: response.data.user.email,
          permission: response.data.user.permission,
          verified: response.data.user.verified,
          jwt: response.data.jwt,
          loggedIn: true,
          response: response
        }
      })
      history.push("/")
      dispatch(setNotification(`Welcome back ${response.data.user.forename}`, "loginSuccess", true))
    }).catch((error) => {
      if(error.response.data === "NOT VERIFIED"){
        const verifyMsg = "Your email has not yet been veirified.  You will need to verify your email before you can log in"
        dispatch(setNotification(verifyMsg, "warning", true))
      } else {
        dispatch({
          type: "SET_USER_LOGGED_IN",
          payload: {
            error: error.response
          }
        });
        dispatch(setNotification("The username or password you entered was incorrect", "error", true));
      }
    })
  }
}

export const logOutUser = () => {
  return {
    type: "LOG_OUT_USER",
    payload: {
      forename: "",
      surname: "",
      email: "",
      permission: "",
      verified: "",
      jwt: "",
      loggedIn: false
    }
  }
}

export const setVerficationProcess = (endpoint, verificationDetails) => {

  return (dispatch) => {

    post(endpoint, verificationDetails).then((response) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "completed",
          token: "",
          error: {}
        }
      })
    }).catch((error) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "not verified",
          token: verificationDetails.token,
          error: error.response
        }
      })
    })
  }
}

export const setRedirect = (status) => {
  return {
    type: "SET_REDIRECT",
    payload: {
      status: status
    }
  }
}

export const logOut = () => {
  return {
    type: "RESET_APP"
  }
}

// export const getUsers = (endpoint, header) => {
//   return {
//     type: "GET_USER",
//     payload: {
//       userId
//   }
// }
