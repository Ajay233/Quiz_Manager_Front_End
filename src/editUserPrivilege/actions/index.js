import { getUsingParams } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const setUserResults = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("users/findByEmail", param, jwt).then((response) => {
      dispatch({
        type: "SET_USER_RESULTS",
        payload: response.data
      })
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        const msg = error.response.data;
        dispatch(setNotification(msg, "error", true));
        console.log(error.response)
      }
    });
  }
}

export const clearUserResults = () => {
  return {
    type: "CLEAR_USER_RESULTS",
    payload: {}
  }
}
