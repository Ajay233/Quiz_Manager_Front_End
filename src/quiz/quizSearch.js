import React from 'react';
import { connect } from 'react-redux'
import Quizes from './quizes'
import QuizSearchByName from '../forms/findQuizByName'
import Notification from '../notifications/notifications'
import { setNotification } from '../notifications/actions'
import { setQuizes, getQuizSearchResults } from './actions'
import "../stylesheets/quizSearch.css"


class QuizSearch extends React.Component {

  renderResultsTitle = (quizes) => {
    return quizes.length === 0 ? null : <div id="resultHeader">{`${quizes.length} Results`}</div>;
  }

  renderResultHeadings = (quizes) => {
    return quizes.length === 0 ? null : this.resultHeadings()
  }

  resultHeadings = () => {
    return(
      <div id="resultsHeadings">
        <div id="nameHeader">Quiz Name</div>
        <div id="descriptionHeader">Description</div>
        <div id="optionsHeader">Options</div>
      </div>
    )
  }

  render(){
    const { quizes, userData, setNotification, getQuizSearchResults } = this.props
    return(
      <div id="quizSearch">
        <Notification />
        <div id="quizSearchTitle">Quiz Search</div>
        <div id="quizSearchByName">
          <QuizSearchByName jwt={userData.jwt} setNotification={setNotification} getQuizSearchResults={getQuizSearchResults} />
        </div>
        <div id="quizSearchByCategory">

        </div>
        <div>
          {this.renderResultsTitle(quizes)}
          {this.renderResultHeadings(quizes)}
          <Quizes quizes={quizes} permission={userData.permission} />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    notificationData: state.notificationData,
    quizes: state.quizes
  }
}

export default connect(mapStateToProps, { setQuizes, getQuizSearchResults, setNotification })(QuizSearch)
