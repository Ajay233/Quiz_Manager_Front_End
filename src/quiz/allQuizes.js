import React from 'react'
import { connect } from 'react-redux'
import Notification from '../notifications/notifications'
import QuizResults from '../quizSearch/quizResults'
import Modal from '../modal/modal'

import { getAllQuizes, deleteQuizFromCategory, clearQuizes } from '../quizSearch/actions'
import { hideModal } from '../modal/actions'
import { setNotification } from '../notifications/actions'
import { getQuestions } from '../question/actions'
import { del } from '../axiosRequests/requests'
import { sessionExpired } from '../utils/session'
import history from '../history'

import "../stylesheets/allQuizzes.css"

export class AllQuizes extends React.Component {

  // componentWillUnmount(){
  //   this.props.clearQuizes();
  // }

  renderModal = () => {
    const { hideModal } = this.props
    const { showModal, showModal2 } = this.props.modalState
    if(showModal){
      return(
        <Modal
          show={showModal}
          title={"Delete Quiz"}
          message={"You are about to delete a quiz which will also delete any questions and answers associated with it.  Do you want to continue?"}
          onDelete={this.handleDelete}
          onCancel={hideModal}
        />
      );
    } else if(showModal2){
      return(
        <Modal
          type={"start"}
          show={showModal2}
          title={"Start Quiz"}
          message={"You are about to start a quiz, would you like to continue?"}
          onStart={this.handleStartQuiz}
          onCancel={hideModal}
        />
      );
    } else {
      return null
    }
  }

  renderCategoryIcon = (category) => {
    switch (category) {
      case "Comics": return <div><img className="categoryIcon" src={require("../public/icons/comics.png")} alt=""/></div>
      case "TV": return <div><img className="categoryIcon" src={require("../public/icons/tv.png")} alt=""/></div>
      case "Gaming": return <div><img className="categoryIcon" src={require("../public/icons/games.png")} alt=""/></div>
      case "Films": return <div><img className="categoryIcon" src={require("../public/icons/films2.png")} alt=""/></div>
      case "Sports": return <div><img className="categoryIcon" src={require("../public/icons/sport.png")} alt=""/></div>
      case "Music": return <div><img className="categoryIcon" src={require("../public/icons/music.png")} alt=""/></div>
      case "History": return <div><img className="categoryIcon" src={require("../public/icons/museumV2.png")} alt=""/></div>
      case "Science": return <div><img className="categoryIcon" src={require("../public/icons/science.png")} alt=""/></div>
      case "Technology": return <div><img className="categoryIcon" src={require("../public/icons/technology.png")} alt=""/></div>
      case "Art": return <div><img className="categoryIcon" src={require("../public/icons/art.png")} alt=""/></div>
      case "Literature": return <div><img className="categoryIcon" src={require("../public/icons/books.png")} alt=""/></div>
      case "General Knowledge": return <div><img className="categoryIcon" src={require("../public/icons/generalKnowledge.png")} alt=""/></div>
      case "Food and Drink": return <div><img className="categoryIcon" src={require("../public/icons/Food-icon.png")} alt=""/></div>
      case "Geography": return <div><img className="categoryIcon" src={require("../public/icons/globe-icon.png")} alt=""/></div>
      default: return null
    }
  }

  renderQuizResults = (filteredQuizList, quizes, listItem, userData, setNotification) => {
    if(filteredQuizList.length > 0){
      return(
        <React.Fragment>
          {this.renderResultHeadings(filteredQuizList)}
          <QuizResults
            key={"Quiz" + quizes.indexOf(listItem)}
            quizes={filteredQuizList}
            permission={userData.permission}
            jwt={userData.jwt}
            setNotification={setNotification}
          />
        </React.Fragment>
      );
    } else {
      return <div id="noQuizzesMsg">There are currenty no quizzes for this category</div>
    }
  }

  // add contitional css alternate between left and right align classes
  // Will need a way to turn this on and off, possibly by using history.location.pathname
  // so this is only applied on the all quizes view
  renderCategories = () => {
    const { quizes, userData, setNotification } = this.props
    let quizesList = quizes.map(listItem => {
      let filteredQuizList = userData.permission === "USER" ? listItem.quizList.filter(quiz => quiz.status === "READY") : listItem.quizList;
      return(
        <div key={"QuizCategory" + quizes.indexOf(listItem)} className="quizCategorySection">
          <div className="categoryHeader">
            <div className="categoryTitle">{listItem.category}</div>
            <div>{this.renderCategoryIcon(listItem.category)}</div>
          </div>
          {this.renderQuizResults(filteredQuizList, quizes, listItem, userData, setNotification)}
        </div>
      );
    })
    return quizesList;
  }

  renderResultHeadings = (quizes) => {
    return quizes.length === 0 ? null : this.resultHeadings()
  }

  resultHeadings = () => {
    const { permission } = this.props.userData
    return(
      <div id="resultsHeadings" className="resultHeadings">
        <div className={permission !== "USER" ? "nameHeader" : "nameHeaderExpanded"}>Quiz Name</div>
        <div className="descriptionHeader">Description</div>
        { permission !== "USER" ? <div className="statusHeader">Status</div> : null }
      </div>
    )
  }

  handleDelete = () => {
    const { quiz, userData, deleteQuizFromCategory, setNotification, hideModal } = this.props;
    const config = {
      data: quiz
    }
    del("quiz/delete", config, userData.jwt).then((response) => {
      hideModal()
      deleteQuizFromCategory(quiz)
      setNotification("Quiz deleted", "success", true)
    }).catch((error) => {
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(this.props.dispatch);
      } else {
        hideModal()
        setNotification("Error - Unable to delete this quiz", "error", true)
      }
    })
  }

  handleStartQuiz = () => {
    const { userData, quiz, getQuestions, hideModal } = this.props
    const param = { quizId: quiz.id }
    getQuestions("question/findByQuizId", param, userData.jwt, true)
    hideModal()
    history.push("/startQuiz")
  }

  renderComponent = () => {
    const { quizes } = this.props;
    if(quizes === null || quizes.length === 0 ){
      //This will need to be replaced with a component that renders an animated loading modal
      //Reuse modal
      return(
        <div>Loading...</div>
      );
    } else {
      return(
        <React.Fragment>
          <Notification />
          <div id="allQuizzesTitle">Browse all quizzes by Category</div>
          {this.renderModal()}
          {this.renderCategories()}
        </React.Fragment>
      );
    }
  }

  render(){
    return(
      <div className="componentContainer">
        {this.renderComponent()}
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    quizes: state.quizes,
    quiz: state.quiz,
    modalState: state.modalState
  }
}

export default connect(mapStateToProps,
  { getAllQuizes,
    deleteQuizFromCategory,
    getQuestions,
    hideModal,
    setNotification,
    clearQuizes
  })(AllQuizes)
