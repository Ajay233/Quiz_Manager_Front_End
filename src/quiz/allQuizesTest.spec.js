import React from 'react'
import ReactDOM from 'react-dom'
import { AllQuizes, mapStateToProps } from './allQuizes'
import { Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import mockAxios from 'jest-mock-axios'
import { render, fireEvent, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { getAllQuizes, deleteQuiz, clearQuizes } from '../quizSearch/actions'
import { hideModal } from '../modal/actions'
import { setNotification } from '../notifications/actions'
import { sessionExpired } from '../utils/session'
import { getQuestions } from '../question/actions'
import history from '../history'

const mockStore = configureStore({})

jest.mock('../quizSearch/actions')
jest.mock('../modal/actions')
jest.mock('../notifications/actions')
jest.mock('../utils/session')
jest.mock('../question/actions')
jest.mock('../history')

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      userData: {
        id: 1
      },
      quizes: [
        { id: 1, name: "test", description: "test" },
        { id: 2, name: "test2", description: "test2" }
      ],
      quiz: { id: 1, name: "test", description: "test" },
      modalState: { showModal: false, showModal2: false, showModal3: false }
    }

    const componentState = mapStateToProps(appState)
    expect(componentState).toEqual(appState)
  })
})


describe("AllQuizes", () => {

  let store;
  let userData;
  let modalState;
  let quizes;
  let quiz;

  beforeEach(() => {
    store = mockStore({
      userData: {
        id: 1,
        permision: "USER"
      },
      quizes: [
        {
          category: "test1",
          quizList: [
            { id: 1, name: "test", description: "test", category: "test1", status: "READY" },
            { id: 2, name: "test2", description: "test2", category: "test1", status: "DRAFT" }
          ]
        },
        {
          category: "test2",
          quizList: [
            { id: 1, name: "test", description: "test", category: "test2", status: "READY" },
            { id: 2, name: "test2", description: "test2", category: "test2", status: "DRAFT" }
          ]
        },
        {
          category: "test3",
          quizList: []
        }
      ],
      quiz: { id: 1, name: "test", description: "test" },
      modalState: { showModal: true, showModal2: false, showModal3: false },
      notificationData: {
        message: "",
        type: "",
        show: false,
        timed: true
      }
    })

    modalState = { showModal: false, showModal2: false, showModal3: false }

    quizes = [
      {
        category: "test1",
        quizList: [
          { id: 1, name: "test", description: "test", category: "test1", status: "READY" },
          { id: 2, name: "test2", description: "test2", category: "test1", status: "DRAFT" }
        ]
      },
      {
        category: "test2",
        quizList: [
          { id: 1, name: "test", description: "test", category: "test2", status: "READY" },
          { id: 2, name: "test2", description: "test2", category: "test2", status: "DRAFT" }
        ]
      },
      {
        category: "test3",
        quizList: []
      }
    ]

    userData = {
      id: 1,
      permision: "USER",
      jwt: "jwt"
    }

    quiz = { id: 1, name: "test", description: "test" }

    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  it("should render only READY quizes for a user", () => {
    const component = renderer.create(
      <Provider store={store}>
        <AllQuizes
          userData={userData}
          modalState={modalState}
          quizes={quizes}
          clearQuizes={clearQuizes}
        />
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  it("should render all quizes regardless of status for ADMIN users", () => {
    userData={
      id: 1,
      permision: "ADMIN"
    }

    const component = renderer.create(
      <Provider store={store}>
        <AllQuizes
          userData={userData}
          modalState={modalState}
          quizes={quizes}
          clearQuizes={clearQuizes}
        />
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  describe("handleDelete", () => {

    const renderModal1 = { showModal: true, showModal2: false, showModal3: false }

    afterEach(() => {
      setNotification.mockClear()
      hideModal.mockClear()
      mockAxios.reset()
    })

    it("should call action creators when a quiz has been deleted", () => {

      const wrapper = render(
        <Provider store={store}>
          <AllQuizes
            userData={userData}
            modalState={renderModal1}
            quizes={quizes}
            clearQuizes={clearQuizes}
            hideModal={hideModal}
            deleteQuiz={deleteQuiz}
            setNotification={setNotification}
          />
        </Provider>
      )

      const requestResponse = {
        data: "DELETED"
      }

      fireEvent.click(wrapper.getByTestId("modal-delete-button"))
      mockAxios.mockResponse(requestResponse)
      expect(hideModal).toHaveBeenCalledTimes(1)
      expect(deleteQuiz).toHaveBeenCalledTimes(1)
      expect(setNotification).toHaveBeenCalledTimes(1)
      expect(setNotification).toHaveBeenCalledWith("Quiz deleted", "success", true)
    })

    it("should call sessionExpired if the error status is 403", () => {

      const wrapper = render(
        <Provider store={store}>
          <AllQuizes
            userData={userData}
            modalState={renderModal1}
            quizes={quizes}
            clearQuizes={clearQuizes}
            hideModal={hideModal}
            deleteQuiz={deleteQuiz}
            setNotification={setNotification}
          />
        </Provider>
      )

      const errorResponse = {
        response: {
          status: 403
        }
      }

      fireEvent.click(wrapper.getByTestId("modal-delete-button"))
      mockAxios.mockError(errorResponse)
      expect(sessionExpired).toHaveBeenCalledTimes(1)
    })

    it("should call setNotification for any other error", () => {

      const wrapper = render(
        <Provider store={store}>
          <AllQuizes
            userData={userData}
            modalState={renderModal1}
            quizes={quizes}
            clearQuizes={clearQuizes}
            hideModal={hideModal}
            deleteQuiz={deleteQuiz}
            setNotification={setNotification}
          />
        </Provider>
      )

      const errorResponse = {
        response: {
          status: 404
        }
      }

      fireEvent.click(wrapper.getByTestId("modal-delete-button"))
      mockAxios.mockError(errorResponse)
      expect(hideModal).toHaveBeenCalledTimes(1)
      expect(setNotification).toHaveBeenCalledTimes(1)
      expect(setNotification).toHaveBeenCalledWith("Error - Unable to delete this quiz", "error", true)
    })
  })

  describe("handleStartQuiz", () => {
    it("should call getQuestions, hideModal, history.push", () => {

      const renderModal2 = { showModal: false, showModal2: true, showModal3: false }

      const wrapper = render(
        <Provider store={store}>
          <AllQuizes
            userData={userData}
            modalState={renderModal2}
            quizes={quizes}
            quiz={quiz}
            clearQuizes={clearQuizes}
            hideModal={hideModal}
            deleteQuiz={deleteQuiz}
            setNotification={setNotification}
            getQuestions={getQuestions}
          />
        </Provider>
      )

      const param = { quizId: 1 }

      fireEvent.click(wrapper.getByTestId("modal-start-button"))
      expect(getQuestions).toHaveBeenCalledTimes(1)
      expect(getQuestions).toHaveBeenCalledWith("question/findByQuizId", param, "jwt", true)
      expect(hideModal).toHaveBeenCalledTimes(1)
      expect(history.push).toHaveBeenCalledTimes(1)
      expect(history.push).toHaveBeenCalledWith("/startQuiz")
    })
  })

})
