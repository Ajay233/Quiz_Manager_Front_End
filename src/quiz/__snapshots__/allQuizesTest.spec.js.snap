// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllQuizes should render all quizes regardless of status for ADMIN users 1`] = `
<div
  className="componentContainer"
>
  <div
    id="allQuizzesTitle"
  >
    Browse all quizzes by Category
  </div>
  <div
    className="quizCategorySection"
  >
    <div
      className="categoryHeader"
    >
      <div
        className="categoryTitle"
      >
        test1
      </div>
      <div />
    </div>
    <div
      id="resultsHeadings"
    >
      <div
        id="nameHeader"
      >
        Quiz Name
      </div>
      <div
        id="descriptionHeader"
      >
        Description
      </div>
    </div>
    <div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test
          </div>
          <div
            className="quizDescription"
          >
            test
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test2
          </div>
          <div
            className="quizDescription"
          >
            test2
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="quizCategorySection"
  >
    <div
      className="categoryHeader"
    >
      <div
        className="categoryTitle"
      >
        test2
      </div>
      <div />
    </div>
    <div
      id="resultsHeadings"
    >
      <div
        id="nameHeader"
      >
        Quiz Name
      </div>
      <div
        id="descriptionHeader"
      >
        Description
      </div>
    </div>
    <div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test
          </div>
          <div
            className="quizDescription"
          >
            test
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test2
          </div>
          <div
            className="quizDescription"
          >
            test2
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="quizCategorySection"
  >
    <div
      className="categoryHeader"
    >
      <div
        className="categoryTitle"
      >
        test3
      </div>
      <div />
    </div>
    <div
      id="noQuizzesMsg"
    >
      "There are currenty no quizzes for this category"
    </div>
  </div>
</div>
`;

exports[`AllQuizes should render only READY quizes for a user 1`] = `
<div
  className="componentContainer"
>
  <div
    id="allQuizzesTitle"
  >
    Browse all quizzes by Category
  </div>
  <div
    className="quizCategorySection"
  >
    <div
      className="categoryHeader"
    >
      <div
        className="categoryTitle"
      >
        test1
      </div>
      <div />
    </div>
    <div
      id="resultsHeadings"
    >
      <div
        id="nameHeader"
      >
        Quiz Name
      </div>
      <div
        id="descriptionHeader"
      >
        Description
      </div>
    </div>
    <div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test
          </div>
          <div
            className="quizDescription"
          >
            test
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test2
          </div>
          <div
            className="quizDescription"
          >
            test2
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="quizCategorySection"
  >
    <div
      className="categoryHeader"
    >
      <div
        className="categoryTitle"
      >
        test2
      </div>
      <div />
    </div>
    <div
      id="resultsHeadings"
    >
      <div
        id="nameHeader"
      >
        Quiz Name
      </div>
      <div
        id="descriptionHeader"
      >
        Description
      </div>
    </div>
    <div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test
          </div>
          <div
            className="quizDescription"
          >
            test
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
      <div
        className="quizContainer"
      >
        <div
          className="quiz"
        >
          <div
            className="quizName"
          >
            test2
          </div>
          <div
            className="quizDescription"
          >
            test2
          </div>
          <div
            className="options"
          >
            <a
              className="start"
              onClick={[Function]}
            >
              <i
                className="far fa-play-circle blue"
              />
               Start
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="quizCategorySection"
  >
    <div
      className="categoryHeader"
    >
      <div
        className="categoryTitle"
      >
        test3
      </div>
      <div />
    </div>
    <div
      id="noQuizzesMsg"
    >
      "There are currenty no quizzes for this category"
    </div>
  </div>
</div>
`;
